File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.cpp
class Solution {
public:
    
    class Function{
        public:
        int BloomDay(vector<int>&plantTime, vector<int>&growTime){
            
            vector<pair<int,int>>vp ; 
            
            int n = plantTime.size()  ; 
            
            for(int i=0;i<n;i++){
                vp.push_back({growTime[i],plantTime[i]})  ; 
            }
            
            sort(vp.begin(),vp.end()) ; 
            
            int sum = 0  ; 
            
            for(auto k : vp){
                sum = max(sum , k.first) + k.second ; 
            }
            return sum ; 
        }
    } ; 
    int earliestFullBloom(vector<int>& plantTime, vector<int>& growTime) {
       
        Function obj ; 
        return obj.BloomDay(plantTime , growTime) ; 
    }
};

File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.java
class Solution {
    public int earliestFullBloom(int[] plantTime, int[] growTime) {
      
        PriorityQueue<int[]>pq = new PriorityQueue<>((a,b)->a[0]-b[0]) ; 
        
        for(int i=0;i<growTime.length ; i++){
            pq.add(new int[]{growTime[i],plantTime[i]})  ; 
        }
        
        int sum = 0 ; 
        while(!pq.isEmpty()){
            int[]t = pq.poll()  ; 
            sum=Math.max(sum,t[0]) + t[1]  ; 
            
        }
        
        return sum ; 
    }
}